components/hab/src/cli.rs

```diff
@@ -1095,15 +1095,25 @@ pub fn sub_svc_status() -> App<'static, 'static> {
     )
 }
 
-fn sub_svc_stop(feature_flags: FeatureFlag) -> App<'static, 'static> {
+pub fn parse_optional_arg<T: FromStr, E>(name: &str, m: &ArgMatches) -> Result<Option<T>, E>
```

Comment:
I think we will only be using this in the case that we've successfully gotten past the validation stage.

In that case, we could tighten up the return type to be simply `Option<T>`, since presumably any input that couldn't be successfully parsed would have been weeded out before now. We could tack an `expect()` to `s.parse()`, and call it a day.

(I've been writing some code elsewhere that deals with these conversions and it's been both clarifying and simplifying to write functions with the knowledge that the input will have been validated beforehand.)